src/graph/graph.c

```diff
@@ -813,6 +813,34 @@ uint Graph_GetNodeLabels
 	return i;
 }
 
+// removes node and all of its connections within the graph
+void Graph_DeleteNode
+(
+	Graph *g,
+	Node *n
+) {
+	// assumption, node is completely detected,
+	// there are no incoming nor outgoing edges
+	// leading to / from node
+	ASSERT(g != NULL);
+	ASSERT(n != NULL);
+
+	RG_Matrix N = Graph_GetNodeLabelMatrix(g);
```

Comment:
consider adding a DEBUG block which checks that node has NO incoming / outgoing edges